### ISSUE TITLE
expand type synonym action 

### ISSUE THREAD

**Issue by MangoIV** [haskell/haskell-language-server#3740]
## Is your enhancement request related to a problem? Please describe.

Sometimes I feel the need to replace a type synonym with its definition; as it's transparent, this should not be an issue 

## Describe the solution you'd like

Add a code action that you can use when having your cursor above a type synonym in a type signature that can replace a type signature with its definition 

## Describe alternatives you've considered

manual replacement is tedious, which is why I would prefer an action 

## Additional context

I can maybe take this on but I never worked on HLS so idk

**Comment by michaelpj** [comment]
Seems like something the `retrie` plugin could support?

**Comment by michaelpj** [comment]
Indeed, looks to me like it might already do that (https://github.com/haskell/haskell-language-server/blob/master/plugins/hls-retrie-plugin/src/Ide/Plugin/Retrie.hs#L467), but I think it only creates code actions at the _declaration_ site. I'm not sure it handles selective inlining of specific instances?

**Comment by michaelpj** [comment]
So I think ultimately this is https://github.com/haskell/haskell-language-server/issues/1758

**Comment by michaelpj** [comment]
Yeah, I'll close as duplicate.
